### PR TITLE
move project fields to PROJECT and remove project/folder prefixes

### LINKED ISSUE #140: Remove project. and folder. prefixes from generated dirs

### PR DESCRIPTION
**Pull request by umairidris**
Fix #140

### CODE REVIEW

File: examples/tfengine/full.yaml [hunk 1]
```diff
@@ -119,15 +121,16 @@ templates:
...
-      HOST_PROJECT_ID: "example-prod-networks"
-      SUBNETS:
-      - NAME: example-subnet
+    PROJECT:
+      PROJECT_ID: "example-prod-apps"
+      APIS:
+      - "container.googleapis.com"
+      SHARED_VPC:
+        HOST_PROJECT_ID: "example-prod-networks"
+        SUBNETS:
```
Comment (xingao267) [inline examples/tfengine/full.yaml:132]: could you remind me if this creates the subnet or not?

File: templates/tfengine/recipes/org/project.yaml [hunk 2]
```diff
@@ -12,14 +12,16 @@
...
-{{$output_path := get . "OUTPUT_PATH" (printf "./project.%s" .PROJECT_ID)}}
+{{$output_path := get . "OUTPUT_PATH" .PROJECT.PROJECT_ID}}
 
 templates:
 - component_path: "../../components/terragrunt/leaf"
   output_path: "{{$output_path}}/project"
 - name: "project"
   component_path: "../../components/project/project"
   output_path: "{{$output_path}}/project"
+  flatten:
```
Comment (xingao267) [inline templates/tfengine/recipes/org/project.yaml:23]: why do we flatten the `PROJECT` here but not in other recipes?

Comment (umairidris) [reply templates/tfengine/recipes/org/project.yaml:23]: None of the other templates in this list require the fields within PROJECT (e.g. they get the project ID through a terragrunt reference), so there was no need to pass in PROJECT keys to them

File: examples/tfengine/full.yaml (hunk 1 above)
Comment (umairidris) [reply examples/tfengine/full.yaml:132]: It only adds permissions on the subnet for this project's servic accounts, the subnet itself is in the network project.

**Pull request merged**